inst/rmd/rnasum.Rmd

```diff
@@ -425,157 +421,99 @@ tmpg <- tmpg |>
-}
+ref_genes.list[["summary"]]$Immune <- RNAsum::immune_summary(
+  tbl_imarkers = ref_genes.list[["genes_immune"]][["genes_immune_markers"]],
+  tbl_igram = ref_genes.list[["genes_immune"]][["genes_immunogram"]],
+  igram_param = params$immunogram
+)
 
 # HRD (homologous recombination deficiency) genes
-ref_genes.list[["summary"]]$HRD <- unique(ref_genes.list[["genes_hrd"]]$SYMBOL)
-
-##### Remove NAs
-ref_genes.list[["summary"]]$HRD <- ref_genes.list[["summary"]]$HRD[!(is.na(ref_genes.list[["summary"]]$HRD))]
+ref_genes.list[["summary"]]$HRD <- ref_genes.list[["genes_hrd"]] |>
+  dplyr::pull("SYMBOL") |>
+  stats::na.omit()
 
 # Cancer genes derived from UMCCR Cancer Gene list (https://github.com/vladsaveliev/NGS_Utils/blob/master/ngs_utils/reference_data/key_genes/umccr_cancer_genes.2019-03-20.tsv) and from OncoKB portal (http://oncokb.org/#/cancerGenes)
-ref_genes.list[["summary"]]$Cancer <- rownames(ref_genes.list[["genes_cancer"]])
-
-##### Remove NAs
-ref_genes.list[["summary"]]$Cancer <- ref_genes.list[["summary"]]$Cancer[!(is.na(ref_genes.list[["summary"]]$Cancer))]
+ref_genes.list[["summary"]]$Cancer <- ref_genes.list[["genes_cancer"]] |>
+  dplyr::pull("Gene") |>
+  unique() |>
+  stats::na.omit()
 
 ##### Record all genes of interest
-genes2keep <- unique(unlist(ref_genes.list[["summary"]]))
+genes2keep <- unlist(ref_genes.list[["summary"]]) |> unique()
 ```
 
 ```{r goi_annotation, comment = NA, message=FALSE, warning=FALSE}
 ##### Get gene symbols for the genes of interest. These genes will not be filtered out due to low/insufficient expression
 
 ##### Get genes genomic coordiantes
-gene_info <- ensembldb::select(edb, keys = keys, columns = c("GENEID", "GENENAME"), keytype = "GENEID")
-names(gene_info) <- gsub("GENEID", "ENSEMBL", names(gene_info))
-names(gene_info) <- gsub("GENENAME", "SYMBOL", names(gene_info))
-
-##### Limit genes annotation to the gene of interest
-genes2keep <- gene_info[gene_info$SYMBOL %in% genes2keep, ]
-
-##### Remove rows with duplicated ENSEMBL IDs
+gene_info <- edb |>
+  ensembldb::select(keys = keys, columns = c("GENEID", "GENENAME"), keytype = "GENEID") |>
+  dplyr::rename("ENSEMBL" = "GENEID", "SYMBOL" = "GENENAME")
+
+##### Limit genes annotation to the gene of interest, then
+##### remove rows with duplicated ENSEMBL IDs
+genes2keep <- gene_info |>
+  dplyr::filter(.data$SYMBOL %in% genes2keep)
 genes2keep <- genes2keep[!duplicated(genes2keep$ENSEMBL), ]
 rownames(genes2keep) <- genes2keep$ENSEMBL
 
 ##### Remove rows with duplicated gene symbols (Y_RNAs, SNORs, LINC0s etc). Preferably select ENSEMBL ID that is used in the count data
-genes2keep.combined_data <- genes2keep[genes2keep$ENSEMBL %in% rownames(ref_dataset.list[[dataset]]$combined_data), ]
-genes2keep <- genes2keep[genes2keep$SYMBOL %!in% genes2keep.combined_data$SYMBOL, ]
+genes2keep.combined_data <- genes2keep |>
+  dplyr::filter(.data$ENSEMBL %in% rownames(ref_dataset.list[[dataset]][["combined_data"]]))
```

Comment:
are we going to keep `rownames` call as last time you mentioned this is error prone?